### PR TITLE
refactor: update webcam "WebRTC MediaMTX" client

### PR DESCRIPTION
**Pull request by meteyou**
## Description

This PR update the "WebRTC MediaMTX" client for MediaMTX v0.23 and above.

## Related Tickets & Documents

fixes: #1559

## Mobile & Desktop Screenshots/Recordings

none

### CODE REVIEW

**Comment by zyrorl** [comment]
git, 'er done.

File: src/components/webcams/WebrtcMediaMTX.vue [hunk 1]
```diff
@@ -87,140 +96,242 @@ export default class WebrtcRTSPSimpleServer extends Mixins(BaseMixin, WebcamMixi
...
+        fetch(this.url, {
+            method: 'POST',
+            headers: {
+                'Content-Type': 'application/sdp',
+            },
+            body: offer.sdp,
         })
+            .then((res) => {
+                if (res.status !== 201) throw new Error('bad status code')
+                this.eTag = res.headers.get('E-Tag')
```
Comment (mryel00) [inline src/components/webcams/WebrtcMediaMTX.vue:240]: This got changed to `ETag` with MediaMTX 1.1.1 as you can see [here](https://github.com/bluenviron/mediamtx/pull/2401)

**Review by mryel00 — APPROVED** [review]

File: src/components/webcams/WebrtcMediaMTX.vue [hunk 2]
```diff
@@ -87,140 +96,246 @@ export default class WebrtcRTSPSimpleServer extends Mixins(BaseMixin, WebcamMixi
...
+            headers: {
+                'Content-Type': 'application/sdp',
+            },
+            body: offer.sdp,
         })
+            .then((res) => {
+                if (res.status !== 201) throw new Error('bad status code')
+                this.eTag = res.headers.get('ETag')
+
+                // fallback for MediaMTX v1.1.0 with broken ETag header
```
Comment (mryel00) [inline src/components/webcams/WebrtcMediaMTX.vue:242]: Correct would be v1.0.x and 1.1.0

File: src/components/webcams/WebrtcMediaMTX.vue [hunk 3]
```diff
@@ -87,140 +96,246 @@ export default class WebrtcRTSPSimpleServer extends Mixins(BaseMixin, WebcamMixi
...
+
+        window.console.log(`[WebRTC mediamtx] ${msg}`)
+    }
+
     // webrtc player methods
-    // adapated from sample player in https://github.com/mrlt8/docker-wyze-bridge
-    start() {
-        // unterminate we're starting again.
-        this.terminated = false
+    // adapated from https://github.com/bluenviron/mediamtx/blob/main/internal/core/webrtc_read_index.html
```
Comment (mryel00) [inline src/components/webcams/WebrtcMediaMTX.vue:109]: Small typo `adapated` instead of `adapated`

**Review by mryel00 — APPROVED** [review]

**Review by mryel00 — APPROVED** [review]

**Pull request merged**